controllers/kubemarkmachine_controller.go

```diff
@@ -508,3 +514,35 @@ func getKubemarkExtendedResourcesFlag(extendedResources infrav1.KubemarkExtended
 	flags := fmt.Sprintf("--extended-resources=%s", strings.Join(resources, ","))
 	return flags
 }
+
+// getKubemarkRegisterWithTaintsFlag returns the raw kubemark command line flag for
+// --registration-with-taints` if they are specified in the spec or an empty string if not.
+// format: comma separated "<key>=<value>:<effect>"
+func getKubemarkRegisterWithTaintsFlag(taints []corev1.Taint) string {
+	if len(taints) == 0 {
+		return ""
+	}
+	taintstrings := []string{}
+	for _, taint := range taints {
+		ts := fmt.Sprintf("%s=%s:%s", taint.Key, taint.Value, taint.Effect)
```

Comment:
Maybe use `taint.ToString()` instead. https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/api/core/v1/taint.go#L28

Reply:
i don't know if it matters, but the serialization from `taint.ToString()` behaves slightly different than the code here.

if we have the taint

```
Key: foo
Effect: NoExecute
```

the current code would serialize as `foo=:NoExecute`

whereas `taint.ToString` will serialize as `foo:NoExecute`

i'll test it out

Reply:
works in testing